test/plugin/test_out_elasticsearch.rb

```diff
@@ -954,10 +1001,10 @@ def test_connection_failed_retry
       raise Faraday::ConnectionFailed, "Test message"
     end
 
-    driver.emit(sample_record)
-
-    assert_raise(Fluent::ElasticsearchOutput::ConnectionFailure) {
-      driver.run
+    assert_raise(Fluent::Plugin::ElasticsearchOutput::ConnectionFailure) {
```

Comment:
So I think I understand now, the driver catches the exception and prints it, as seen here: https://github.com/fluent/fluentd/blob/master/lib/fluent/test/driver/base.rb#L141-L150. I don't understand why this behaviour is desirable, or how we can work around it.